src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserService.cs

```diff
@@ -132,90 +132,137 @@ public void InitializeService(ServiceHost serviceHost)
             FileBrowserCloseParams fileBrowserParams,
             RequestContext<FileBrowserCloseResponse> requestContext)
         {
-            FileBrowserCloseResponse response = new FileBrowserCloseResponse();
-            FileBrowserOperation removedOperation;
-            response.Succeeded = ownerToFileBrowserMap.TryRemove(fileBrowserParams.OwnerUri, out removedOperation);
-
-            if (removedOperation != null && this.expandNodeQueue != null)
+            try
             {
-                bool hasPendingQueueItems = this.expandNodeQueue.HasPendingQueueItems;
-                if (removedOperation.FileTreeCreated && !hasPendingQueueItems)
-                {
-                    removedOperation.Dispose();
-                    this.expandNodeQueue.CloseConnections(removedOperation.SqlConnection.DataSource, removedOperation.SqlConnection.Database, DefaultExpandTimeout);
-                }
-                else if (!removedOperation.FileTreeCreated)
-                {
-                    removedOperation.Cancel();
-                }
-                else if (hasPendingQueueItems)
-                { 
-                    this.expandNodeQueue.StopQueueProcessor(DefaultExpandTimeout);
-                }
+                var task = Task.Run(() => RunFileBrowserCloseTask(fileBrowserParams, requestContext))
+                    .ContinueWithOnFaulted(null);
+                await requestContext.SendResult(new FileBrowserCloseResponse() { Succeeded = true });
+            }
+            catch
+            {
+                await requestContext.SendResult(new FileBrowserCloseResponse());
```

Comment:
>requestContext [](start = 22, length = 14)

should we log the failure or include it in the return response?

Reply:
updated to log the failure